### PR TITLE
Support for apiserver disabled AuthN and AuthZ

### PR DESCRIPTION
**Pull request by samrocketman**
Description
-----------

This change is a UI update for when the backend apiserver has authentication and authorization disabled via properties.

The backend apiserver can potentially disable its own authentication and authorization allowing anonymous full access.  This is desirable for a local dependencytrack stack meant for local only development.

Addressed Issue
---------------

closes https://github.com/DependencyTrack/frontend/issues/333

Additional Details
------------------

Unfortunately, disabling authN and authZ on apiserver appears to set some kind of readonly mode.  I'm not able to upload SBoMs.

This is my first time developing NodeJS.  I built the app for my own testing the following way.  My host is Ubuntu 20.04 Linux.

```bash
docker run --rm -it \
  -u "$(id -u):$(id -g)" \
  -v "$PWD:$PWD" \
  -w "$PWD" \
  node:16 /bin/bash -exc 'npm install; npm run build'

docker build -t frontend:snapshot -f docker/Dockerfile.alpine .
```

Checklist
---------

- [x] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/github-templates/CONTRIBUTING.md#pull-requests)
- [x] This PR introduces new or alters existing behavior, and I have updated the [documentation](https://github.com/DependencyTrack/dependency-track/tree/master/docs/_docs) accordingly

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -20,6 +20,7 @@ import api from "./shared/api.json";
 import oidc from "./shared/oidc.json";
 import version from "./version";
 import { getContextPath } from "./shared/utils";
+import EventBus from './shared/eventbus';
```
Comment (samrocketman) [inline src/main.js:23]:
```suggestion
```

Remove unused import (from an earlier experiment of my while attempting this feature).

**Comment by stevespringett** [comment]
Although Java Alpine supports anonymous mode, the Dependency-Track API server does not. There are currently no plans on supporting anonymous mode. However, if the community wants this capability, many changes will be necessary in the API to support this.

See also: https://github.com/DependencyTrack/dependency-track/issues/653

**Comment by samrocketman** [comment]
> Although Java Alpine supports anonymous mode, the Dependency-Track API server does not. There are currently no plans on supporting anonymous mode. However, if the community wants this capability, many changes will be necessary in the API to support this.
> 
> See also: [DependencyTrack/dependency-track#653](https://github.com/DependencyTrack/dependency-track/issues/653)

Thanks for your response.  I am currently developing the backend now and have it WIP.  Working on some validation before I publish a PR.

**Comment by samrocketman** [comment]
@stevespringett here's my proposed backend changes https://github.com/DependencyTrack/dependency-track/pull/2169

**Comment by samrocketman** [comment]
# Some screenshots

Screenshot taken at 2022-11-19_20-21-28

And once I got API server and frontend snapshot versions up and running I decided to upgrade my dependencies to address vulnerabilities.

Screenshot taken at 2022-11-19_21-01-19

Here's my dev container https://github.com/samrocketman/jervis/blob/1a65855f3555990296a7986ad3416a4dd66e31d0/.devcontainer/docker-compose.yml#L88-L115

**Comment by samrocketman** [comment]
Closing per discussion in https://github.com/DependencyTrack/dependency-track/pull/2169

**Pull request closed without merging**